### PR TITLE
Publishing builds from travis to artifactory.broadinstitute.org.

### PR DESCRIPTION
**Pull request by kshakir**
Builds from the `develop` branch will be published as git labeled snapshots.
Tagged builds will be deployed as releases.

### CODE REVIEW

**Comment by kshakir** [comment]
Tested manually against a named branch and with tags.

File: .travis.yml [hunk 1]
```diff
@@ -4,6 +4,12 @@ scala:
   - 2.11.7
 jdk:
   - oraclejdk8
-
 script: sbt clean coverage test
-after_success: sbt 'set unmanagedSourceDirectories in Compile := Seq(file("."))' coveralls
+after_success:
+  - sbt coveralls
+  - src/bin/publish_snapshot.sh
+deploy:
```
Comment (scottfrazer) [inline .travis.yml]: I'm not sure I understand how/when this `deploy` section gets invoked

File: src/bin/publish_release.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+echo "TRAVIS_TAG='$TRAVIS_TAG'"
+
+sbt \
+    'set test in Test := {}' \
+    "set version := \"${TRAVIS_TAG}\"" \
+    'set isSnapshot := false' \
+    'set resolvers += Resolver.url("bintray-sbt-plugin-releases", url("http://dl.bintray.com/content/sbt/sbt-plugin-releases"))(Resolver.ivyStylePatterns)' \
+    'set publishTo := Option("artifactory-publish" at "https://artifactory.broadinstitute.org/artifactory/libs-release-local;build.timestamp=" + new java.util.Date().getTime)' \
+    "set credentials += Credentials(\"Artifactory Realm\", \"artifactory.broadinstitute.org\", \"${ARTIFACTORY_USERNAME}\", \"${ARTIFACTORY_PASSWORD}\")" \
```
Comment (scottfrazer) [inline src/bin/publish_release.sh]: How does `ARTIFACTORY_USERNAME` get set?

File: .travis.yml [hunk 3]
```diff
@@ -4,6 +4,12 @@ scala:
...
+  provider: script
+  script: src/bin/publish_release.sh
+  on:
+    tags: true
```
Comment (kshakir) [inline .travis.yml]: > …when…

Runs whenever someone [tags a build](http://docs.travis-ci.com/user/deployment/script/).

File: .travis.yml [hunk 4]
```diff
@@ -4,6 +4,12 @@ scala:
...
   - oraclejdk8
-
 script: sbt clean coverage test
-after_success: sbt 'set unmanagedSourceDirectories in Compile := Seq(file("."))' coveralls
+after_success:
+  - sbt coveralls
+  - src/bin/publish_snapshot.sh
+deploy:
+  provider: script
+  script: src/bin/publish_release.sh
```
Comment (kshakir) [inline .travis.yml]: > …how…

Uses this [script](http://docs.travis-ci.com/user/deployment/script/) to deploy releases.

File: .travis.yml (hunk 3 above)
Comment (scottfrazer) [inline .travis.yml]: can one set `branch: master` as well as `tags: true`?

Comment (scottfrazer) [inline .travis.yml]: Maybe it doesn't matter...

File: src/bin/publish_release.sh (hunk 2 above)
Comment (kshakir) [inline src/bin/publish_release.sh]: Travis [encrypted environment variables](http://docs.travis-ci.com/user/environment-variables/#Defining-Variables-in-Repository-Settings) in the [settings](https://travis-ci.org/broadinstitute/lenthall/settings).

File: .travis.yml (hunk 3 above)
Comment (kshakir) [inline .travis.yml]: One could build on `master` alternatively. I was paranoid about two different people accidentally pushing to master. While merging to master is a manual operation, tagging imho is has even more intent.

File: src/bin/publish_release.sh (hunk 2 above)
Comment (scottfrazer) [inline src/bin/publish_release.sh]: I should just RTFM before I ask a lot of questions :)

**Comment by scottfrazer** [comment]
Second reviewer @mcovarr

File: .travis.yml (hunk 3 above)
Comment (scottfrazer) [inline .travis.yml]: gotcha

File: src/bin/publish_snapshot.sh [hunk 5]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+set -e
+
+echo "TRAVIS_BRANCH='$TRAVIS_BRANCH'"
+echo "TRAVIS_PULL_REQUEST='$TRAVIS_PULL_REQUEST'"
+
+if [ "$TRAVIS_BRANCH" == "develop" ] && [ "$TRAVIS_PULL_REQUEST" == "false" ]; then
+    sbt \
+        'set test in Test := {}' \
```
Comment (mcovarr) [inline src/bin/publish_snapshot.sh]: we really want to publish snapshots with potentially failing tests?

Comment (kshakir) [inline src/bin/publish_snapshot.sh]: Only invoked [`after_success`](https://github.com/broadinstitute/lenthall/pull/2#discussion-diff-41996488R8). No need to wait for them to run again.

**Comment by scottfrazer** [comment]
:+1:

**Comment by mcovarr** [comment]
:+1:

**Comment by kshakir** [comment]
Cool, squashing and merging.

**Pull request merged**